debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -300,6 +312,55 @@ protected void readChunk() throws InterruptedException {
         }
     }
 
+    private boolean isTableInvalid() {
+        final TableId currentTableId = (TableId) context.currentDataCollectionId();
+        currentTable = databaseSchema.tableFor(currentTableId);
+        if (currentTable == null) {
+            LOGGER.warn("Schema not found for table '{}', known tables {}", currentTableId, databaseSchema.tableIds());
+            nextDataCollection();
+            return true;
+        }
+        if (currentTable.primaryKeyColumns().isEmpty()) {
+            LOGGER.warn("Incremental snapshot for table '{}' skipped cause the table has no primary keys", currentTableId);
+            nextDataCollection();
+            return true;
+        }
+        return false;
+    }
+
+    private void verifySchema() throws SQLException {
+        if (context.isSchemaVerified()) {
+            return;
+        }
+        if (context.getSchema() == null) {
+            context.setSchema(readSchema());
+        }
+        else {
+            if (!context.isSchemaVerified()) {
+                IncrementalSnapshotSchema currentSchema = readSchema();
+                if (context.getSchema().equals(currentSchema)) {
+                    context.setSchemaVerified(true);
+                }
+                else {
+                    context.setSchema(currentSchema);
```

Comment:
Why is the verified flag not set to `false` here?

Reply:
Refactored this method